### PR TITLE
add trust Cypress tests and supporting POs

### PR DESCRIPTION
**Pull request by CMurrell148**
_Added Trust page automation and AIT sub page automation_

https://dev.azure.com/dfe-gov-uk/Academies-and-Free-Schools-SIP/_workitems/edit/177209/

## Checklist

- [x] Pull request attached to the appropriate user story in Azure DevOps
- [x] Testing complete - all manual and automated tests pass

### CODE REVIEW

File: tests/NewCypress/cypress/pages/trusts/trustDetailsPage.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+class TrustDetailsPage {
+
+    elements = {
+        detailsHeader: () => cy.get('header > .govuk-heading-l'),
+        trustDetailSummaryCard: () => cy.get(':nth-child(2) > .govuk-summary-card'),
+        detailsSummaryCardContentBox: () => cy.get(':nth-child(2) > .govuk-summary-card > .govuk-summary-card__content'),
+        referenceNumberSummaryCard: () => cy.get(':nth-child(3) > .govuk-summary-card'),
+        referenceNumberSummaryCardContentBox: () => cy.get(':nth-child(3) > .govuk-summary-card > .govuk-summary-card__content'),
+
```
Comment (dynamictulip) [inline tests/NewCypress/cypress/pages/trusts/trustDetailsPage.ts:9]: Many of these selectors are brittle, when content moves on the page they will break and it won't necessarily be obvious

File: tests/NewCypress/cypress/pages/trusts/governancePage.ts [hunk 2]
```diff
@@ -136,38 +136,38 @@ class GovernancePage {
...
-        this.checkSorting(HistoricMembers.To, HistoricMembers.ToHeader, HistoricMembers.ToHeaderButton);
+        this.checkGovernanceSorting(HistoricMembers.Names, HistoricMembers.NameHeader, HistoricMembers.NameHeaderButton);
+        this.checkGovernanceSorting(HistoricMembers.Roles, HistoricMembers.RoleHeader, HistoricMembers.RoleHeaderButton);
+        this.checkGovernanceSorting(HistoricMembers.AppointedBys, HistoricMembers.AppointedHeader, HistoricMembers.AppointedHeaderButton);
+        this.checkGovernanceSorting(HistoricMembers.From, HistoricMembers.FromHeader, HistoricMembers.FromHeaderButton);
+        this.checkGovernanceSorting(HistoricMembers.To, HistoricMembers.ToHeader, HistoricMembers.ToHeaderButton);
     }
 
-    private checkSorting(elements: () => Cypress.Chainable<JQuery<HTMLElement>>, header: () => Cypress.Chainable<JQuery<HTMLElement>>, headerButton: () => Cypress.Chainable<JQuery<HTMLElement>>) {
+    private checkGovernanceSorting(elements: () => Cypress.Chainable<JQuery<HTMLElement>>, header: () => Cypress.Chainable<JQuery<HTMLElement>>, headerButton: () => Cypress.Chainable<JQuery<HTMLElement>>) {
```
Comment (dynamictulip) [inline tests/NewCypress/cypress/pages/trusts/governancePage.ts:170]: I wonder whether we could move this into a common place and reuse it for table sorting on other pages?

File: tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
+class AcademiesInTrustPage {
+
+    elements = {
+        DetailsPage: {
+            table: () => cy.get('.govuk-main-wrapper > .dfe-width-container > .govuk-grid-row > .govuk-grid-column-three-quarters'),
```
Comment (dynamictulip) [inline tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts:5]: This doesn't select the table, it selects all the content around it too. you could use `cy.get('table)` instead

File: tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            authTypes: () => cy.get('[data-testid="type-of-establishment"]'),
```
Comment (dynamictulip) [inline tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts:6]: this isn't selecting anything to do with auth (or local authorities?)

File: tests/NewCypress/cypress/e2e/regression/trusts/academiesInTrust.cy.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+describe("Testing the components of the Academies in this trust page", () => {
+
+    describe("Details tab", () => {
+        beforeEach(() => {
+            cy.login()
+            cy.visit('/trusts/academies/details?uid=5712')
+        });
+
+        it("Checks the correct AIT details page headers are present", () => {
```
Comment (dynamictulip) [inline tests/NewCypress/cypress/e2e/regression/trusts/academiesInTrust.cy.ts:11]: AIT is not a common acronym. Is it needed at all given that we're in "Testing the components of the Academies in this trust page" > "Details tab"?

File: tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        this.elements.OfstedPage.table().should('contain', 'School name');
+        this.elements.OfstedPage.table().should('contain', 'Pupils eligible for free school meals');
+        this.elements.OfstedPage.table().should('contain', 'Local authority average');
+        this.elements.OfstedPage.table().should('contain', 'National average');
+        return this;
+    }
+
+    public checkAuthTypesOnAITDetailsTable() {
+        this.elements.DetailsPage.authTypes().each(element => {
+            expect(element.text().trim()).to.be.oneOf(["Academy sponsor led", "Academy converter", "University technical college", "Free schools"])
```
Comment (dynamictulip) [inline tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts:58]: These are not local authorities

Comment (CMurrell148) [reply tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts:58]: These were supposed to be school types and not auth types - I must have had a brain moment when naming this and the element

Now updated to reflect school type not auth type

File: tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts (hunk 4 above)
Comment (CMurrell148) [reply tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts:6]: These were supposed to be school types and not auth types - I must have had a brain moment when naming this and the element

Now updated to reflect school type not auth type

File: tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts (hunk 3 above)
Comment (CMurrell148) [reply tests/NewCypress/cypress/pages/trusts/academiesInTrustPage.ts:5]: Now updated to use cy.get('table')

File: tests/NewCypress/cypress/e2e/regression/trusts/academiesInTrust.cy.ts (hunk 5 above)
Comment (CMurrell148) [reply tests/NewCypress/cypress/e2e/regression/trusts/academiesInTrust.cy.ts:11]: Removed AIT from all of the titles - My thinking was to differentiate from the stuff for the details page outside of academies and trusts but I think that's pretty redundant

File: tests/NewCypress/cypress/pages/trusts/governancePage.ts (hunk 2 above)
Comment (CMurrell148) [reply tests/NewCypress/cypress/pages/trusts/governancePage.ts:170]: I'm working on filtering for the next iteration which should be done soon

On my local branch this is done in a new POM/Class called SortingUtility

**Review by dynamictulip — APPROVED** [review]

**Review by dynamictulip — APPROVED** [review]

**Pull request merged**